cfdm/cellconnectivity.py

```diff
@@ -0,0 +1,359 @@
+
+        Returns a description of all properties, including those of
+        components, and provides selected values of all data arrays.
+
+        .. versionadded:: (cfdm) UGRIDVER
+
+        :Parameters:
+
+            display: `bool`, optional
+                If False then return the description as a string. By
+                default the description is printed.
+
+        :Returns:
+
+            {{returns dump}}
+
+        """
+        if _title is None:
+            _title = "Cell Connectivity: " + self.identity(default="")
+
+        return super().dump(
+            display=display,
+            _key=_key,
+            _omit_properties=_omit_properties,
+            _level=_level,
+            _title=_title,
+            _axes=_axes,
+            _axis_names=_axis_names,
+        )
+
+    def identity(self, default=""):
+        """Return the canonical identity.
+
+        By default the identity is the first found from the following,
+        in order:
+
+        * The connectivity, preceded by ``'connectivity:'``.
+        * The ``standard_name`` property.
+        * The ``cf_role`` property, preceded by 'cf_role='.
+        * The ``long_name`` property, preceded by 'long_name='.
+        * The netCDF variable name, preceded by 'ncvar%'.
+        * The value of the default parameter.
+
+        .. versionadded:: (cfdm) UGRIDVER
+
+        .. seealso:: `identities`
+
+        :Parameters:
+
+            default: optional
+                If no identity can be found then return the value of the
+                default parameter.
+
+        :Returns:
+
+                The identity.
+
+        """
+        n = self.get_connectivity(None)
+        if n is not None:
+            return f"connectivity:{n}"
+
+        n = self.get_property("standard_name", None)
+        if n is not None:
+            return n
+
+        for prop in ("cf_role", "long_name"):
+            n = self.get_property(prop, None)
+            if n is not None:
+                return f"{prop}={n}"
+
+        n = self.nc_get_variable(None)
+        if n is not None:
+            return f"ncvar%{n}"
+
+        return default
+
+    def identities(self, generator=False, **kwargs):
+        """Return all possible identities.
+
+        The identities comprise:
+
+        * The connectivity type, preceded by ``'connectivity:'``.
+        * The ``standard_name`` property.
+        * All properties, preceded by the property name and a colon,
+          e.g. ``'long_name:Air temperature'``.
+        * The netCDF variable name, preceded by ``'ncvar%'``.
+
+        .. versionadded:: (cfdm) UGRIDVER
+
+        .. seealso:: `identity`
+
+        :Parameters:
+
+            generator: `bool`, optional
+                If True then return a generator for the identities,
+                rather than the default, a list.
+
+            kwargs: optional
+                Additional configuration parameters. Currently
+                none. Unrecognised parameters are ignored.
```

Comment:
Good point. I think it's more "reserved for use in subclasses". How about:
```suggestion
            kwargs: optional
                Additional configuration parameters that may be required by
                subclasses. Unrecognised parameters are ignored.
```
Looking at how it's all used in cfdm and cf-python, there doesn't seem to be much point in it, so perhaps kwargs should be removed anyway? But that's a conversation for another place.